pkg/metrics/workqueue.go

```diff
@@ -95,8 +95,18 @@ func init() {
 	Registry.MustRegister(unfinished)
 	Registry.MustRegister(longestRunningProcessor)
 	Registry.MustRegister(retries)
+}
+
+// SetDefaultGlobalWorkqueueMetricsProvider registers the GlobalWorkqueueMetricsProvider
+// as the global workqueue.MetricsProvider.
+func SetDefaultGlobalWorkqueueMetricsProvider() {
+	workqueue.SetProvider(GlobalWorkqueueMetricsProvider())
+}
 
-	workqueue.SetProvider(workqueueMetricsProvider{})
```

Comment:
As mentioned in the k/k pr, we wont merge this. This breaks the majority use-case (`workqueue metrics just work without doing anything`) in order to support a minority use-case (`set custom provider`).

/hold

Reply:
Is this not what you were suggesting here? https://github.com/kubernetes/kubernetes/pull/114242#issuecomment-1468802756

This is marked as a breaking change, and allows consumers to set it. Unsure what the path forward is if it is neither of the options I've proposed.

Reply:
No, my suggestion was to change the upstream workqueue to not use a sync.Once so what is set by controller-runtime can be overridden later if desired.

Reply:
Alright, I'll send a PR for that -- seems to paper over the fact that this library is mutating global state and the "metrics just work without doing anything" could be addressed somewhere else (like kubebuilder, which sets up a `main`), but if this gets it in 🤷🏼 

I'll leave this open in the meantime if others want to weigh in.

Reply:
I think changing kubebuilder does not really mitigate the situation. I don't have a good overview over the ecosystem but the controllers that I'm aware of don't regualarly regenerate their main file with kubebuilder.

Reply:
Okie, I'll close this then -- I've also got a PR for what Alvaro suggested here, if you guys want to drive that approach: https://github.com/kubernetes/kubernetes/pull/116616